modules/storages/app/services/storages/project_storages/create_service.rb

```diff
@@ -29,5 +29,26 @@
 # See: ../storages/create_service.rb for comments on services
 module Storages::ProjectStorages
   class CreateService < ::BaseServices::Create
+    protected
+
+    def after_perform(service_call)
+      super(service_call)
+
+      project_storage = service_call.result
+
+      return service_call if project_storage.project_folder_mode.to_sym == :inactive
+
+      last_project_folder_result =
+        LastProjectFolderPersistenceHelper.create_last_project_folder(
+          user:,
+          projects_storage_id: project_storage.id,
+          origin_folder_id: project_storage.project_folder_id,
+          mode: project_storage.project_folder_mode
+        )
+
+      service_call.add_dependent!(last_project_folder_result)
+
+      service_call
```

Comment:
I think `.tap` is a good choice here. Would you like it?
```suggestion
      service_call.tap do |service_call|
        if project_storage.project_folder_mode.to_sym != :inactive
          service_call.add_dependent!(
            LastProjectFolderPersistenceHelper.create_last_project_folder(
              user:,
              projects_storage_id: project_storage.id,
              origin_folder_id: project_storage.project_folder_id,
              mode: project_storage.project_folder_mode
            )
          )
        end
      end
```

Reply:
used tap, but reduced nesting a bit ;)

Reply:
disregard that, raised a short question in `dev` channel, and got back a big bikeshedding discussion, with the result, that we want to avoid using `.tap` just for sparing the return line.